### ISSUE TITLE
Image shows too stretched out in the image popup

### ISSUE THREAD

**Issue by annjawn** [Orderella/PopupDialog#147]
What is the image size requirements for the image popup? I can't seem to find that anywhere. I tried to include some standard size images but the popup either becomes too large or the image gets scaled up so it looks blurry. I feel like there is lower limit to the image size - can someone point me to what it is?

**Comment by gmogames** [comment]
I'm running into the same problem. I need to include an icon like a checkmark or an X for error that are 60x60 or smaller, but the image stretches to a very large size. Not sure how to make it smaller just by using uiimage and not having access to the imageview

![image](https://user-images.githubusercontent.com/1765046/27505439-a48b837a-5854-11e7-8955-4ab7ed9f339f.png)

**Comment by annjawn** [comment]
I ended up taking the image (the @2x one) from the resource folder of the sample project and taking it's size in Photoshop and then resized my own image with similar dimensions. And now it's showing up correctly. I had the same issue as yours, but as of now with the edited image it shows properly in iPhone 6s Plus, 6s, 5, and SE. Less than ideal but works.

![22](https://user-images.githubusercontent.com/26977294/27506236-64559eb6-5869-11e7-8f30-ed93ee4a0fe5.png)

![33](https://user-images.githubusercontent.com/26977294/27506239-6a8bd070-5869-11e7-8fe3-0033d89f87df.png)

**Comment by mwfire** [comment]
Hey @annjawn, @gmogames,

The image provided to the default view controller will take the full width of the dialog. The height, on the other hand, is calculated in order to maintain aspect ratio. This prevents weird stretching effects.

If you want the image to be centered, for now you have to add blank space around the image. This way, almost any use case is covered. For special cases you can create your own custom view controller.

In future versions, the imageView of the default popup view will be publicly accessible, so you can access contentmode.

At the moment, @annjawn's solution is the way to go.

Cheers
Martin

**Comment by annjawn** [comment]
@mwfire 
Thanks. I actually did start with a custom view controller (using a xib file) and a normal image was showing fine in that but it looks like the custom popup has a max height. I know there is a max width based on the documentation but I am not sure if a custom view controller popup has a max height requirement. I have this custom xib created with the showed dimension, but the actual popup kept cutting off after the first line i.e. the OK button shows immediately below it hiding the entire paragraph of text both in simulator and on device. Since, that didn't work, I resorted to using the built in image popup, but had to do the image sizing myself in Photoshop.

I kept the width of the VC similar to what the width is on the xib file in sample project for the "Ratings" popup, but had to give it more height to accommodate the text. I was hoping it would fit, but for some reason it didn't and kept showing a max height similar to the "Ratings" popup in the sample project. 🙁

By the way, the image, the first line of text, and the second paragraph are all inside a stack view pinned to the top, bottom, left and right of the VC.

![44](https://user-images.githubusercontent.com/26977294/27511907-9927702a-58e5-11e7-8de0-23a7a5145519.png)
